Make `MinLength` required on `GenerateParameterDefault`

afscrome opened the pull request:
## Description

Default `GenerateParameterDefault` password to 128 bits of entropy

Previously if you forgot to specify a `MinLength`, you would end up with a password of length zero which is a bit surprising and dangerous.  Set a default value that will give 128 bits of entropy.

Fixes #10570

## Checklist

- Is this feature complete?
  - [X] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [X] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue (consider using one of the following templates):
      - [New (or update) `doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)
      - [New `breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)
      - [New `diagnostic` template](https://github.com/dotnet/docs-aspire/issues/new?template=06-diagnostic-addition.yml)
  - [X] No

JamesNK commented:
I think setting a default value would be better. Defaulting the min length to one that generates over 128 bits of entrothy (assuming other default settings are unchanged) would help create more secure apps by default. Also, it leaves open the option to generate zero length values, which I assume there are valid use cases for it in some apps.

Thoughts @davidfowl @mitchdenny?

afscrome committed:
Default `GenerateParameterDefault` password to 128 bits of entropy

Previously if you forgot to specify a `MinLength`, you would end up with a password of length zero which is a bit surprising and dangerous.  Set a default value that will give 128 bits of entropy.

Fixes #10570
